pkg/plugins/golang/v4/scaffolds/internal/templates/api/spoke.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/kubebuilder/v4/pkg/machinery"
+)
+
+var _ machinery.Template = &Spoke{}
+
+// Spoke scaffolds the file that defines spoke version conversion
+type Spoke struct {
+	machinery.TemplateMixin
+	machinery.MultiGroupMixin
+	machinery.BoilerplateMixin
+	machinery.ResourceMixin
+
+	Force bool
+}
+
+// SetTemplateDefaults implements file.Template
+func (f *Spoke) SetTemplateDefaults() error {
+	if f.Path == "" {
+		if f.MultiGroup && f.Resource.Group != "" {
+			f.Path = filepath.Join("api", "%[group]", f.Resource.Webhooks.Spoke, "%[kind]_conversion.go")
+		} else {
+			f.Path = filepath.Join("api", f.Resource.Webhooks.Spoke, "%[kind]_conversion.go")
+		}
+	}
+
+	f.Path = f.Resource.Replacer().Replace(f.Path)
+	log.Println(f.Path)
+
+	f.TemplateBody = spokeTemplate
+
+	if f.Force {
+		f.IfExistsAction = machinery.OverwriteFile
+	} else {
+		f.IfExistsAction = machinery.SkipFile
+	}
+
+	return nil
+}
+
+// nolint:lll
+const spokeTemplate = `{{ .Boilerplate }}
+
+package {{ .Resource.Webhooks.Spoke }}
+
+import (
+	"log"
+
+    "sigs.k8s.io/controller-runtime/pkg/conversion"
+    {{ .Resource.ImportAlias }} "{{ .Resource.Path }}"
+)
+
+// ConvertTo converts this {{ .Resource.Kind }} to the Hub version ({{ .Resource.Version }}).
+func (src *{{ .Resource.Kind }}) ConvertTo(dstRaw conversion.Hub) error {
+    dst := dstRaw.(*{{ .Resource.ImportAlias }}.{{ .Resource.Kind }})
+	log.Printf("Converting from %T to %T", dst.APIVersion, src.APIVersion)
```

Comment:
```suggestion
	log.Printf("Converting from %T to %T", src.APIVersion, dst.APIVersion)
```

IIUC it's the other way around - `ConverTo` converts to the hub version to store it in the API server. 
Eg: https://github.com/kubeflow/kubeflow/blob/ec82fbf58b79cff529d948b96e44ffd06bdfe679/components/notebook-controller/api/v1alpha1/notebook_conversion.go#L24